NKL/Encode.lean

```diff
@@ -0,0 +1,331 @@
+/-
```

Comment:
Would it make sense to have a more user friendly version of this as either a either a big comment or separate doc?

Reply:
I am not sure what you mean. A more friendly version of what?

Reply:
Oh, just the formatter. I was imagining a bnf-looking sort of thing without all the monads. But maybe it's fine to not have it in a doc for now.